pvanalytics/features/daytime.py

```diff
@@ -0,0 +1,210 @@
+    # flag the full day for substitution. This may result in slightly
+    # reduced accuracy for sunrise/sunset times on these days (even if
+    # one end of the day - sunrise or sunset - was correctly flagged,
+    # it will be replaced with the rolling median for that minute).
+    day_length = night.groupby(night.cumsum()).transform(
+        lambda x: len(x) * minutes_per_value
+    )
+    # remove night time values so they don't interfere with the median
+    # day length.
+    day_length.loc[night] = np.nan
+    day_length_median = day_length.rolling(window='14D').median()
+    # flag days that are more than 30 minutes shorter than the median
+    short_days = day_length < (day_length_median - 30)
+    invalid = short_days.groupby(short_days.index.date).transform(
+        lambda day: any(day)
+    )
+    return _smooth_if_invalid(night, invalid)
+
+
+def _filter_and_normalize(series, outliers):
+    # filter a series by removing outliers and clamping the minimum to
+    # 0. Then normalize the series by the maximum deviation.
+    if outliers is not None:
+        series.loc[outliers] = np.nan
+    series.loc[series < 0] = 0
+    return (series - series.min()) / (series.max() - series.min())
+
+
+def _freqstr_to_minutes(freqstr):
+    return pd.to_timedelta(
+        frequencies.to_offset(freqstr)
+    ).seconds / 60
+
+
+def diff(series, outliers=None, low_value_threshold=0.003,
+         low_median_threshold=0.0015, low_diff_threshold=0.0005,
+         clipping=None, freq=None):
+    """Return True for values that are during the day.
+
+    After removing outliers and normalizing the data, periods of
+    low-slope and low-value are identified as night. A time is
+    classified as night when two of the following three criteria are
+    satisfied:
+
+    - near-zero value
+    - near-zero first-order derivative
+    - near-zero rolling median at the same time over the surrounding
+      week
+
+    It is possible that mid-day times where power goes near zero or
+    stops changing can be incorrectly classified as night. To correct
+    these errors times where the total duration of periods marked
+    night or day is too long or too short are identified and values at
+    these times are compared to values at the same time for the
+    preceding two weeks and following two weeks. The day/night value
+    is adjusted to match the median value of these days.
+
+    Finally any values that are True in `clipping` are marked as day.
+
+    Parameters
+    ----------
+    series : Series
+        Time series of power or irradiance.
+    outliers : Series, optional
+        Boolean time series with True for values in `series` that are
+        outliers.
+    low_value_threshold : float, default 0.003
+        Maximum normalized value for a time to be considered night.
+    low_median_threshold : float, default 0.0015
+        Minimum rolling median for a time to be considered night.
```

Comment:
The rolling median should be low at night. This should be a maximum threshold
```suggestion
        Maximum rolling median for a time to be considered night.
```